editor/scene_tree_dock.cpp

```diff
@@ -386,6 +389,9 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 			if (!profile_allow_editing) {
 				break;
 			}
+			if (!_validate_no_foreign()) {
```

Comment:
Previously this would fail silently instead, so this only adds a warning without preventing anything new. Although is useful for #69087

Reply:
It's been updated now to be inside the 'is_anything_selected' check, but yes, it doesn't prevent anything new, but it does make its behaviour consistent with the other actions in providing explicit feedback when a rename check fails.